crates/solidity/outputs/npm/tests/src/cst/rewriter.test.mts

```diff
@@ -0,0 +1,109 @@
+import assert from "node:assert";
+import {
+  BaseRewriter,
+  NonterminalNode,
+  TerminalNode,
+  TerminalKind,
+  NonterminalKind,
+  EdgeLabel,
+  Edge,
+  Node,
+} from "@nomicfoundation/slang/cst";
+import { Parser } from "@nomicfoundation/slang/parser";
+import { LanguageFacts } from "@nomicfoundation/slang/utils";
+
+class BasicRewriter extends BaseRewriter {
+  insideContract = false;
+
+  public override rewriteIdentifier(terminalNode: TerminalNode): Node | undefined {
+    if (!this.insideContract) {
+      return TerminalNode.create(terminalNode.kind, terminalNode.unparse() + "_new");
+    } else {
+      return terminalNode;
+    }
+  }
+
+  public override rewriteContractDefinition(node: NonterminalNode): Node | undefined {
+    this.insideContract = true;
+    const newChildren = node.children().map((edge) => {
+      if (edge.label == EdgeLabel.Name) {
+        return Edge.createWithTerminal(EdgeLabel.Name, TerminalNode.create(TerminalKind.Identifier, "NewName"));
```

Comment:
IIUC, `rewriteContractDefinition()` stops the recursion at `ContractDefinition`, without calling `rewriteIdentifier()`.. Are they related? If not, I suggest separating each rewriter with its tests..

```ts
describe("Foo Rewriter", () => {
  class FooRewriter extends BaseRewriter { }

  it("....", () => { });
  
  it("....", () => { });
})